File: dataset/output/CLAUDE_claude-3-5-sonnet-20240620/Task20_CLAUDE_claude-3-5-sonnet-20240620.java

public class DigPow {
    public static long digPow(int n, int p) {
        String digits = String.valueOf(n);
        long sum = 0;
        for (int i = 0; i < digits.length(); i++) {
            int digit = Character.getNumericValue(digits.charAt(i));
            sum += Math.pow(digit, p + i);
        }
        if (sum % n == 0) {
            return sum / n;
        }
        return -1;
    }

    public static void main(String[] args) {
        System.out.println(digPow(89, 1));
        System.out.println(digPow(92, 1));
        System.out.println(digPow(695, 2));
        System.out.println(digPow(46288, 3));
    }
}

File: dataset/output/CLAUDE_claude-3-5-sonnet-20240620/Task20_CLAUDE_claude-3-5-sonnet-20240620.cpp

#include <cmath>
#include <string>

class DigPow {
public:
    static long long digPow(int n, int p) {
        long long sum = 0;
        std::string num = std::to_string(n);
        for (char c : num) {
            sum += std::pow(c - '0', p++);
        }
        return sum % n == 0 ? sum / n : -1;
    }
};
